src/Orleans/Statistics/ClientStatisticsManager.cs

```diff
@@ -93,7 +93,8 @@ internal ClientStatisticsManager(IStatisticsConfiguration config)
                 }
                 else if (config.UseAzureSystemStore)
                 {
-                    var statsDataPublisher = await StatsTableDataManager.GetManager(false, config.DataConnectionString, config.DeploymentId, transport.MyAddress.Endpoint.ToString(), clientId.ToString(), config.DNSHostName);
+                    var statsDataPublisher = await StatsTableDataManager.GetManager(false, config.DataConnectionString, config.DeploymentId,
```

Comment:
That one line is the actual fix (clientId.ToParsableString instead of clientId.ToString). The rest of changes are just cleanup and better diagnostics.